### DISCUSSION TITLE
Unable to Connect on iOS Browser

### DISCUSSION THREAD

**Discussion by MuckT** [WalletConnect/walletconnect-monorepo#3549] [Developer Support]
### Description

It appears I am unable to initialize the chat client on using an iOS mobile browser such as: Safari, Edge, or Chrome; Connecting via Scanning a QR code and via Android does not appear to have the same issues. If I use the backup 'Still doesn't work? Try this alternate link' the connection is successful; however, the final 'allow this app to sync' action request is not served.

### Reproduction Steps
1. On iOS in edge or chrome on a non private browser clean the cache and navigate to `chore://inspect` or `edge://inspect`.
2. Tap Start Logging.
3. In a new window navigate to a website where the chat client is initialized on connection and attempt to connect with a wallet.
4. Observe errors and the WalletConnect Modal persisting.

### First Pairing Attempt
```Bash
...
LOG Sign Client Initialize End
LOG Chat Client Initialize Start 
ERROR[object Object] [HistoryClient] Failed to register tags: {}
...
```
### Second Pairing Attempt
```Bash
...
LOG Sign Client Initialize End
LOG Chat Client Initialize Start 
LOG Chat Client Initialize End
ERROR[object Object] [object Object] Failed to decode message from topic: 'b0de42ed04efc501217b86ed12c0b8dc61d8b8875fbbbab08de23d6c598580e7', clientId: 'did:key:<...>'
ERROR[object Object] [object Object] Error: No matching key. keychain: b0de42ed04efc501217b86ed12c0b8dc61d8b8875fbbbab08de23d6c598580e7
ERROR[object Object] [object Object] Failed to decode message from topic: 'c44c91003addd730e0808d71651072bc732a3902c6ba332e7d44b1f846511d14', clientId: 'did:key:<...>'
ERROR[object Object] [object Object] Error: No matching key. keychain: c44c91003addd730e0808d71651072bc732a3902c6ba332e7d44b1f846511d14
ERROR[object Object] Error: Missing or invalid. Decoded payload on topic c44c91003addd730e0808d71651072bc732a3902c6ba332e7d44b1f846511d14 is not identifiable as a JSON-RPC request or a response.
```

### Versions
```JSON
"@walletconnect/chat-client": "^0.7.4",
"@walletconnect/core": "^2.10.0",
"@walletconnect/sign-client": "^2.10.0",
"@walletconnect/sync-client": "^0.3.7",
"@web3modal/ethereum": "^2.7.1",
"@web3modal/react": "2.7.1",
"@web3modal/standalone": "^2.4.3", 
"viem": "^1.7.0",
"wagmi": "^1.3.10",
```

**Comment by yucheng-Li** [comment]
Same problem!!!

**Comment by Vineetkg20** [comment]
"It seems like the issue is related to how keys are managed in the iOS browser environment. You could try debugging by ensuring that the keychain is properly initialized before attempting the connection. Additionally, verifying the browser's WebSocket connection stability might help. You may also want to check if the latest version of WalletConnect libraries resolves this, as iOS-specific issues are often addressed in patches.

**Comment by yucheng-Li** [comment]
已收到
